### PR TITLE
Include ordering for constraints on babelfish tables

### PR DESCRIPTION
**Pull request by rishabhtanwar29**
### Description
Babelfish supports column ordering and NULLs ordering in table constraints but PG
does not. It was implemented through special grammar rule and few dialect T-SQL dialect
dependent logic in engine. Currently this is not handled in pg_dump so the information
about column/nulls ordering gets lost during dump. Although, this information is available
in the underlying index for a UNIQUE/PRIMARY KEY constraint but this information anyway
gets lost since the underlying index does not get dumped for a UNIQUE/PRIMARY KEY constraint,
instead it is dumped as a constraint.

To fix this issue, we will now dump the underlying index of a UNIQUE/PRIMARY KEY constraint
and make use of ALTER TABLE ADD CONSTRAINT USING snytax to add that as constraint.
Postgres disallows creation of an constraint using index like this for non default sort ordering as
native pg_dump does not handle such constraints. So we will need to bypass that check for
Babelfish dump/restore with this special handling in bbf_dump.

Task: BABEL-4888
Signed-off-by: Rishabh Tanwar <ritanwar@amazon.com>

Engine PR: https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/395

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/JDBC/expected/BABEL-4888-vu-verify.out [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+~~ERROR (Code: 515)~~
+
+~~ERROR (Message: null value in column "id1" of relation "babel_4888_t4" violates not-null constraint)~~
+
+
+INSERT INTO babel_4888_t4 VALUES (NULL, NULL, 0x9999)
+GO
+~~ERROR (Code: 515)~~
+
+~~ERROR (Message: null value in column "id1" of relation "babel_4888_t4" violates not-null constraint)~~
```
Comment (surendravishnoi) [inline test/JDBC/expected/BABEL-4888-vu-verify.out:129]: we should also validate both asc and desc ordering, null first and last ordering

Comment (rishabhtanwar29) [reply test/JDBC/expected/BABEL-4888-vu-verify.out:129]: Actually these combinations are already being tested. If you check the prepare file, table `babel_4888_t1` for example, has `id1` in `DESC` order and `id2` in `ASC` order.

Comment (surendravishnoi) [reply test/JDBC/expected/BABEL-4888-vu-verify.out:129]: but how are we validating that NULLs are actually at right places and index have right ordering?

Comment (rishabhtanwar29) [reply test/JDBC/expected/BABEL-4888-vu-verify.out:129]: We are verifying it with the index definition, check the SELECT command below.

**Review by sumitj824 — APPROVED** [review]

**Review by shardgupta — APPROVED** [review]

**Pull request merged**